tests/unit/test_endpoints.py

```diff
@@ -294,3 +295,26 @@ def __init__(self, service, old_endpoints, new_endpoints):
 
     def run(self):
         assert_equal(self.old_endpoints, self.new_endpoints)
+
+
+def test_no_lost_endpoints():
+    # This makes sure that a bad sync doesn't cause us to loose any services
+
+    data_dir = os.path.join(os.path.dirname(__file__), 'data')
+    old_endpoints_file = os.path.join(data_dir, 'old_endpoints.json')
+
+    with open(old_endpoints_file) as f:
+        old_endpoints = json.load(f)
+
+    with open(boto.ENDPOINTS_PATH) as f:
+        new_endpoints = json.load(f)
+
+    builder = StaticEndpointBuilder(BotoEndpointResolver(new_endpoints))
+    built = builder.build_static_endpoints()
+
+    # Assert no services are lost
+    for service, service_endpoints in old_endpoints.items():
+        new_service_endpoints = built.get(service, {})
+        for region, regional_endpoint in service_endpoints.items():
+            new_regional_endpoint = new_service_endpoints.get(region)
+            yield assert_equal, regional_endpoint, new_regional_endpoint
```

Comment:
Might be good to have a more explicit error message that mentions what service and region was being used on fail for this assertion.